hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/main/java/org/apache/hadoop/yarn/server/federation/utils/FederationStateStoreFacade.java

```diff
@@ -376,10 +378,23 @@ public void updateApplicationHomeSubCluster(
    */
   public SubClusterId getApplicationHomeSubCluster(ApplicationId appId)
       throws YarnException {
-    GetApplicationHomeSubClusterResponse response =
-        stateStore.getApplicationHomeSubCluster(
+    try {
+      if (isCachingEnabled()) {
+        Object value =
+            cache.get(buildGetApplicationHomeSubClusterRequest(appId));
+        if (value instanceof SubClusterId) {
+          return (SubClusterId) value;
+        } else {
+          throw new YarnException("Cannot be converted to SubClusterId.");
```

Comment:
Maybe add the type of the original.

Reply:
This part of the code does need cast, I simplified the code.